### PR TITLE
Some code compactings

### PR DESCRIPTION
**Pull request by eumiro**
A few independent code compactings.

### CODE REVIEW

File: fpdf/fonts.py [hunk 1]
```diff
@@ -1,14 +1,9 @@
...
 
-fpdf_charwidths = {"courier": {}}
-
-for i in range(256):
-    fpdf_charwidths["courier"][chr(i)] = 600
-    fpdf_charwidths["courierB"] = fpdf_charwidths["courier"]
-    fpdf_charwidths["courierI"] = fpdf_charwidths["courier"]
-    fpdf_charwidths["courierBI"] = fpdf_charwidths["courier"]
+fpdf_charwidths = {
+    name: {chr(i): 600 for i in range(256)}
```
Comment (Lucas-C) [inline fpdf/fonts.py:4]: Why making this change?
With the new version, the code is less readable...
Sometimes a simple `for` loop is easier to read 😊

File: fpdf/fpdf.py [hunk 2]
```diff
@@ -563,7 +562,7 @@ def add_font(self, family, style="", fname=None, uni=False):
         if not fname:
             fname = family.replace(" ", "") + f"{style.lower()}.pkl"
         style = "".join(sorted(style.upper()))
-        if any(letter not in "BI" for letter in style):
+        if set(style) - {"B", "I"}:
```
Comment (Lucas-C) [inline fpdf/fpdf.py:565]: Again, this may be more "compact", but not necessarily more readable...

File: fpdf/fpdf.py [hunk 3]
```diff
@@ -488,12 +488,11 @@ def line(self, x1, y1, x2, y2):
...
 
-    def _set_dash(self, dash_length=False, space_length=False):
+    def _set_dash(self, dash_length=None, space_length=None):
+        dash = ""
         if dash_length and space_length:
-            s = f"[{dash_length * self.k:.3f} {space_length * self.k:.3f}] 0 d"
-        else:
-            s = "[] 0 d"
-        self._out(s)
+            dash = f"{dash_length * self.k:.3f} {space_length * self.k:.3f}"
```
Comment (Lucas-C) [inline fpdf/fpdf.py:494]: 👍

File: fpdf/fpdf.py [hunk 4]
```diff
@@ -698,7 +697,7 @@ def set_font(self, family=None, style="", size=0):
             )
             style = ""
         style = "".join(sorted(style.upper()))
-        if any(letter not in "BIU" for letter in style):
+        if set(style) - {"B", "I", "U"}:
```
Comment (Lucas-C) [inline fpdf/fpdf.py:700]: Same remark on readability

File: fpdf/template.py [hunk 5]
```diff
@@ -117,21 +117,17 @@ def __getitem__(self, name):
                 # text for this page:
                 return self.texts[self.pg_no][key]
             # find first element for default text:
-            elements = [
-                element for element in self.elements if element["name"].lower() == key
-            ]
-            if elements:
-                return elements[0]["text"]
+            return next((x for x in self.elements if x["name"].lower() == key), None)
```
Comment (Lucas-C) [inline fpdf/template.py:120]: Where did the `["text"]` went??

File: test/fonts/test_set_font.py [hunk 6]
```diff
@@ -43,7 +43,7 @@ def test_set_builtin_font(tmp_path):
     pdf = FPDF()
     pdf.add_page()
     builtin_fonts = sorted(
-        f for f in pdf.core_fonts if not any(f.endswith(s) for s in ("B", "I", "BI"))
+        f for f in pdf.core_fonts if not f.endswith(("B", "I", "BI"))
```
Comment (Lucas-C) [inline test/fonts/test_set_font.py:46]: 👍

**Review by Lucas-C — CHANGES_REQUESTED** [review]

File: fpdf/fonts.py (hunk 1 above)
Comment (eumiro) [reply fpdf/fonts.py:4]: The difference is, that in the original case the four subdicts were the same subdict references four times, which has probably no impact when read-only. Anyway, the three subdicts were created at each iteration. At least I'll create them only once. See my new commit.

File: fpdf/template.py (hunk 5 above)
Comment (eumiro) [reply fpdf/template.py]: Lost in translation. Sorry, thanks for the hint. Fixed.

**Comment by eumiro** [comment]
Review notes changed, repushed.

But I have one question, @Lucas-C : 

https://github.com/PyFPDF/fpdf2/blob/cf6a9663702518b720f32d6f8279a2f87644143a/fpdf/template.py#L116-L118

Here we check whether `key` is in `self.texts`, but then access the element BELOW `self.pg_no`. This smells like a bug to me. What do you think?

**Review by alexanderankin — APPROVED** [review]
who knows what template.py is doing

**Comment by alexanderankin** [comment]
if the templating thing is actually a bug, or if that part should be somehow changed thats probably not within the scope of a generic pr like this

**Comment by Lucas-C** [comment]
Thanks for the report @eumiro !
I agree with @alexanderankin about fixing this in another PR.
I'm going to try to add a test that covers this part of the code.

**Comment by eumiro** [comment]
Thank you. Just for the records, I wrote #98.

**Pull request merged**